### PR TITLE
Switch to white if white_value in payload

### PR DESCRIPTION
**Pull request by r4nd0mbr1ck**
I have added code to switch to white mode if the payload contains white_value

This is for Issue #100

### CODE REVIEW

**Comment by sidoh** [comment]
Awesome, thanks for taking a look at this.

`white_value` is a metric, not a boolean, right? The semantics are a little unclear, but I think it amounts to `brightness` restricted to dedicated white LEDs:

https://home-assistant.io/components/light/

I think other HASS milight integrations (and perhaps all other colored bulbs in HASS) switch to white mode when a color close to white on the color picker is selected. Is that not the case in your experience?

**Comment by r4nd0mbr1ck** [comment]
Yep, white_value is a metric. 

I only have LIFX bulbs - which switch to white when I change the color temperature (they have tunable whites). I will test this with LIFX a bit more later tonight but I haven't yet noticed them switching to white via the color picker.

**Comment by r4nd0mbr1ck** [comment]
Did a bit of investigating in the HA codebase. Looks like different light platforms do different things

1) LimitlessLED - uses a white 'effect' and also using a color close to white min(color) > 256 - boundary. So two ways of switching to white. I am just worried that using the color picker may not be accurate enough.

2) LifX - switches to white when color_temp is set. This wouldn't work for the RGBW bulbs as they don't support color temp.

I haven't managed to find how this works for others - lightify, tplink, etc. They seem to be native to the bulbs and/or the python libraries being used. 

P.S.: I have also written code to use the "effect" to set night_mode on the CCT bulbs. I might send that through a PR later.  Perhaps I should include set_white also as an effect irrespective of how we go with implementing the current issue.

**Comment by r4nd0mbr1ck** [comment]
I think I agree that using the RGB would be a better way to go rather than introduce a parameter that doesn't serve it's origibal intent. Will try and work on that.

**Comment by sidoh** [comment]
The effect route sounds great too, especially if it's something the existing LimitlessLED integration already does. 

Btw - changing color temp for milight will also switch RGB+CCT bulbs to white mode.

Thanks again for looking into this :)

File: lib/MiLight/MiLightClient.h [hunk 1]
```diff
@@ -11,7 +11,8 @@
 //#define DEBUG_PRINTF
 
 #define MILIGHT_DEFAULT_RESEND_COUNT 10
-
+//Used to determine close to white
+#define RGB_BOUNDARY 40
```
Comment (sidoh) [inline lib/MiLight/MiLightClient.h]: Not a big deal, but could we call this `RGB_WHITE_BOUNDARY` or something along those lines?

File: lib/MiLight/MiLightClient.cpp [hunk 2]
```diff
@@ -267,16 +267,23 @@ void MiLightClient::update(const JsonObject& request) {
     uint8_t r = color["r"];
     uint8_t g = color["g"];
     uint8_t b = color["b"];
+    //If close to white
+    if( r > 256 - RGB_BOUNDARY && g > 256 - RGB_BOUNDARY && b > 256 - RGB_BOUNDARY)
```
Comment (sidoh) [inline lib/MiLight/MiLightClient.cpp]: Would you mind updating the styling? e.g.:

```
if (...) {
  ...
} else {
  ...
}
```

to be consistent with the rest of the project.

**Comment by sidoh** [comment]
Looks good! Thanks for taking care of this. Were you able to test out the changes?

**Comment by r4nd0mbr1ck** [comment]
I have pushed two commits through - one to fix the code styles and the other to handle effects. 

I have tested the white effect against a RGBW bulb and that appears to work. However, night mode doesn't seem to work. Does the RGBW bulb have a night mode?

Night mode works with the CCT bulb.

**Comment by sidoh** [comment]
Looks great to me. Thanks again!

**Comment by sidoh** [comment]
This is in [1.5.0-dev3](https://github.com/sidoh/esp8266_milight_hub/releases/tag/1.5.0-dev3) now. :)

**Pull request merged**